docs/tasks/administer-cluster/running-cloud-controller.md

```diff
@@ -1,34 +1,85 @@
+can develop their features independantly from [Kubernetes core](https://github.com/kubernetes/kubernetes).
 
-The `cloud-controller-manager` can be linked to any cloud provider that satisifies the [cloudprovider.Interface](https://git.k8s.io/kubernetes/pkg/cloudprovider/cloud.go).
-In future Kubernetes releases, cloud vendors should link code that satisfies the above interface to the `cloud-controller-manager` project and compile `cloud-controller-manager` for their own clouds. Cloud providers would also be responsible for maintaining and evolving their code.
+{% include templates/glossary/snippet.md term="cloud-controller-manager" length="long" %}
+{% endcapture %}
+
+{% capture body %}
+
+## Cloud Controller Manager
+
+Kubernetes v1.6 introduced a new binary called `cloud-controller-manager`. `cloud-controller-manager` is a daemon that embeds cloud-specific control loops. These cloud-specific control loops were originally in the `kube-controller-manager`. Since cloud providers develop and release at a different pace compared to the Kubernetes project, abstracting the provider-specific code to the `cloud-controller-manager` binary allows cloud provider vendors to evolve independently from the core Kubernetes code.
+
+The `cloud-controller-manager` can be linked to any cloud provider that satisifies [cloudprovider.Interface](https://git.k8s.io/kubernetes/pkg/cloudprovider/cloud.go). For backwards compatibility, the [cloud-controller-manager](https://github.com/kubernetes/kubernetes/tree/master/cmd/cloud-controller-manager) provided in the core Kubernetes project uses the same cloud libraries as `kube-controller-manager`. Cloud providers already supported in Kubernetes core are expected to use the in-project cloud-controller-manager to transition out of Kubernetes core. In future Kubernetes releases, all cloud controller managers will be developed outside of the core Kubernetes project managed by sig leads or cloud vendors.
 
 * TOC
 {:toc}
 
-### Building cloud-controller-manager for your cloud
+## Administration
 
-To build cloud-controller-manager for your cloud, follow these steps:
-
-1. Write a cloudprovider that satisfies the [cloudprovider.Interface](https://git.k8s.io/kubernetes/pkg/cloudprovider/cloud.go).
-2. Link the cloudprovider to cloud-controller-manager.
+### Requirements
 
-The methods in [cloudprovider.Interface](https://git.k8s.io/kubernetes/pkg/cloudprovider/cloud.go) are self-explanatory. All of the
-[existing providers](https://git.k8s.io/kubernetes/pkg/cloudprovider/providers) satisfy this interface. If your cloud is already a part
-of the existing providers, you do not need to write a new provider; you can proceed directly with linking your cloud provider to the `cloud-controller-manager`.
+Every cloud has their own set of requirements for running `cloud-controller-manager`, it should not be too different from the requirements when running `kube-controller-manager`. As a general rule of thumb you'll need:
```

Comment:
their own cloud provider integration (instead of `cloud-controller-manager`)

Reply:
Describe this as it should behave as a normal Deployment that can be `kubectl apply`-d to the cluster